platform/frontend/src/components/chat/chat-messages.tsx

```diff
@@ -54,8 +64,97 @@ export function ChatMessages({
   hideToolCalls = false,
   status,
   isLoadingConversation = false,
+  onMessagesUpdate,
+  onUserMessageEdit,
 }: ChatMessagesProps) {
   const isStreamingStalled = useStreamingStallDetection(messages, status);
+  // Track editing by messageId-partIndex to support multiple text parts per message
+  const [editingPartKey, setEditingPartKey] = useState<string | null>(null);
+  const [editingMessageId, setEditingMessageId] = useState<string | null>(null);
+
+  const handleStartEdit = (partKey: string, messageId?: string) => {
+    setEditingPartKey(partKey);
+    if (messageId) {
+      setEditingMessageId(messageId);
+    }
+  };
+
+  const handleCancelEdit = () => {
+    setEditingPartKey(null);
+    setEditingMessageId(null);
+  };
+
+  const handleSaveAssistantMessage = async (
+    messageId: string,
+    partIndex: number,
+    newText: string,
+  ) => {
+    try {
+      // Call the API to update the message (no subsequent deletion for assistant messages)
+      const { data, error } = await updateChatMessage({
+        path: { id: messageId },
+        body: { partIndex, text: newText },
+      });
+
+      if (error) {
+        console.error("Update message API error:", error);
+        const errorMessage =
+          (error as { error?: { message?: string } })?.error?.message ||
+          JSON.stringify(error);
+        throw new Error(`Failed to update message: ${errorMessage}`);
+      }
+
+      // Update local state to reflect the change immediately
+      if (onMessagesUpdate && data?.messages) {
+        onMessagesUpdate(data.messages as UIMessage[]);
+      }
+    } catch (error) {
+      console.error("Failed to update assistant message:", error);
+      throw error;
+    }
+  };
+
+  const handleSaveUserMessage = async (
+    messageId: string,
+    partIndex: number,
+    newText: string,
+  ) => {
+    try {
+      // Call the API to update the message and delete subsequent messages
+      const { data, error } = await updateChatMessage({
+        path: { id: messageId },
+        body: {
+          partIndex,
+          text: newText,
+          deleteSubsequentMessages: true,
+        },
+      });
+
+      if (error) {
```

Comment:
Once you create Tanstack Query mutation, you don't need to handle HTTP error. We're about to apply the global solution for all of those.

And in components we should not have any `try...catch` blocks

Reply:
so is it already in main I can use it ?